src/imitation/rewards/reward_nets.py

```diff
@@ -318,6 +324,134 @@ def forward(self, state, action, next_state, done):
         return outputs
 
 
+class NormalizedRewardNet(RewardNetWrapper):
+    """A reward net that normalizes the output of its base network."""
+
+    def __init__(
+        self,
+        base: RewardNet,
+        normalize_output_layer: Type[nn.Module],
+    ):
+        """Initialize the NormalizedRewardNet.
+
+        Args:
+            base: a base RewardNet
+            normalize_output_layer: The class to use to normalize rewards. This
+                can be any nn.Module that preserves the shape; e.g. `nn.Identity`,
+                `nn.LayerNorm`, or `networks.RunningNorm`.
+        """
+        # FIXME(yawen): by default, the reward output is squeezed to produce
+        # tensors with `(batch_size,`) shape for predict_processed. This works
+        # for networks.RunningNorm, but not for nn.BatchNorm1d that requires
+        # shape of (N,C).
+        super().__init__(base=base)
+        # Assuming reward is scalar, norm layer should be initialized with shape (1,).
+        self.normalize_output_layer = normalize_output_layer(1)
+
+    def predict_processed(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+        update_norm_stats: bool = True,
+    ) -> np.ndarray:
+        """Compute normalized rewards for a batch of transitions without gradients.
+
+        Args:
+            state: Current states of shape `(batch_size,) + state_shape`.
+            action: Actions of shape `(batch_size,) + action_shape`.
+            next_state: Successor states of shape `(batch_size,) + state_shape`.
+            done: End-of-episode (terminal state) indicator of shape `(batch_size,)`.
+            update_norm_stats: Whether to update the running stats of the norm layer.
```

Comment:
Do we actually need this? Originally I thought we did as we had a use-case in preference comparison where, during training a reward learning algorithm, we wanted to call `predict_processed` but not update the stats (when gathering extra samples). But I think that's not needed: since we don't use the learned reward then (*only* for RL training), we can just use the `venv` *before* wrapping.

So now the only use case I can see where we want to call `predict_processed` but not update the stats is for actually evaluating the learned policy. But then it seems natural to put the network into evaluation mode, in which case the stats won't be updated.

So, I now lean towards getting rid of this flag, unless there's another use case in mind for it.